### PR TITLE
Catching exception when failing to read config in partial trust

### PR DESCRIPTION
**Pull request by stimms**
Fix Issue #416  .

Expand the area under try catch to allow for client initialization in partial trust scenarios, specifically as part of a Dynamics 365 plugin. 

- [x ] I ran Unit Tests locally.

### CODE REVIEW

**Comment by stimms** [comment]
I'd love some insight as to why the build failed on this trivial change but the devops builds aren't publicly accessible.

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 1]
```diff
@@ -44,9 +44,17 @@ public PlatformImplementation()
...
         public string ReadConfigurationXml()
         {
-            // Config file should be in the base directory of the app domain
-            string configFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "ApplicationInsights.config");
-
+            string configFilePath;
+            try
+            {
+                // Config file should be in the base directory of the app domain
+                configFilePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "ApplicationInsights.config");
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: can you instead move this line to the already existing try..catch blocks?

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 2]
```diff
@@ -44,9 +44,17 @@ public PlatformImplementation()
...
+            }
+            catch (SecurityException)
```
Comment (cijothomas) [inline src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: can Path.Combine throw SecurityException?

Comment (stimms) [reply src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: I don't believe Path.Combine is the culprit here so much as `AppDomain.CurrentDomain.BaseDirectory`

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 1 above)
Comment (stimms) [reply src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: Not elegantly as the `configFilePath` is referenced in the other exceptions.

File: src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 2 above)
Comment (cijothomas) [reply src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: right. thanks for clarifying.

Comment (stimms) [reply src/Microsoft.ApplicationInsights/Extensibility/Implementation/Platform/PlatformImplementation.cs]: I dug into this a bit more in the reference source and found that getting the `BaseDirectory` exercises https://referencesource.microsoft.com/#mscorlib/system/AppDomainSetup.cs,056ca0529857f8f8 which is indeed what I'm seeing in my stack traces from Dynamics 

```
Unhandled exception: 
Exception type: System.ServiceModel.FaultException`1[Microsoft.Xrm.Sdk.OrganizationServiceFault]
Message: System.Security.SecurityException: Request for the permission of type 'System.Security.Permissions.FileIOPermission, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' failed.
   at System.Security.CodeAccessSecurityEngine.Check(Object demand, StackCrawlMark& stackMark, Boolean isPermSet)
   at System.Security.CodeAccessPermission.Demand()
   at System.AppDomainSetup.VerifyDir(String dir, Boolean normalize)
   at Microsoft.ApplicationInsights.Extensibility.Implementation.Platform.PlatformImplementation.ReadConfigurationXml()
   at Microsoft.ApplicationInsights.Extensibility.Implementation.TelemetryConfigurationFactory.Initialize(TelemetryConfiguration configuration, TelemetryModules modules)
   at Microsoft.ApplicationInsights.Extensibility.TelemetryConfiguration.get_Active()
   at LicenseManager.Services.LoggingService..ctor(String loggerName)
   at LicenseManager.UpdateInterceptor..ctor()
```

**Comment by cijothomas** [comment]
@stimms The build is failing with stylecop checks. Can you build locally in Release, and you will be able to see the same errors.

**Comment by cijothomas** [comment]
There is this test failure, about the new Warning trace.



Test method Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.CoreEventSourceTest.MethodsAreImplementedConsistentlyWithTheirAttributes threw exception: 
System.Exception: ApplicationInsightsConfigNotAccessibleWarning is implemented incorrectly: EventId. expected: '45', actual: '0' ---> System.Exception: EventId. expected: '45', actual: '0'

Stack trace
    at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.AssertEqual[T](T expected, T actual, String message) in \Test\TestFramework\Shared\EventSourceTest.cs:line 115
   at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.VerifyEventId(MethodInfo eventMethod, EventWrittenEventArgs actualEvent) in \Test\TestFramework\Shared\EventSourceTest.cs:line 76
   at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.VerifyMethodImplementation(EventSource eventSource, MethodInfo eventMethod) in \Test\TestFramework\Shared\EventSourceTest.cs:line 34
--- End of inner exception stack trace ---
    at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.VerifyMethodImplementation(EventSource eventSource, MethodInfo eventMethod) in \Test\TestFramework\Shared\EventSourceTest.cs:line 41
   at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.MethodsAreImplementedConsistentlyWithTheirAttributes(EventSource eventSource) in \Test\TestFramework\Shared\EventSourceTest.cs:line 18
   at Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.CoreEventSourceTest.MethodsAreImplementedConsistentlyWithTheirAttributes() in \Test\Microsoft.ApplicationInsights.Test\Shared\Extensibility\Implementation\Tracing\CoreEventSourceTest.cs:line 14

**Comment by cijothomas** [comment]
Failed   MethodsAreImplementedConsistentlyWithTheirAttributes
Error Message:
 Test method Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing.CoreEventSourceTest.MethodsAreImplementedConsistentlyWithTheirAttributes threw exception: 
System.Exception: ApplicationInsightsConfigNotAccessibleWarning is implemented incorrectly: EventId. expected: '45', actual: '0' ---> System.Exception: EventId. expected: '45', actual: '0'
Stack Trace:
    at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.AssertEqual[T](T expected, T actual, String message) in /mnt/a/1/s/Test/TestFramework/Shared/EventSourceTest.cs:line 115
   at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.VerifyMethodImplementation(EventSource eventSource, MethodInfo eventMethod) in /mnt/a/1/s/Test/TestFramework/Shared/EventSourceTest.cs:line 35
--- End of inner exception stack trace ---
    at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.VerifyMethodImplementation(EventSource eventSource, MethodInfo eventMethod) in /mnt/a/1/s/Test/TestFramework/Shared/EventSourceTest.cs:line 41
   at Microsoft.ApplicationInsights.TestFramework.EventSourceTest.MethodsAreImplementedConsistentlyWithTheirAttributes(EventSource eventSource) in /mnt/a/1/s/Test/TestFramework/Shared/EventSourceTest.cs:line 16

**Pull request merged**